dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/AlertDao.java

```diff
@@ -244,6 +244,13 @@ private void saveTaskTimeoutAlert(Alert alert, String content, int alertGroupId)
     public void sendTaskTimeoutAlert(WorkflowInstance workflowInstance,
                                      TaskInstance taskInstance,
                                      ProjectUser projectUser) {
+        assert projectUser != null;
```

Comment:
```suggestion
        if(projectUser == null) {
                throw IlleganArguementException("project user is null");
        }
```

`assert` will do nothing if the jvm doesn't enable it.

And I don't think we need to check the project user here, since this shouldn't affect the alert, alert only need to take care about alert plugin, if the project user is null, it only affects the alert content.

Reply:
> if(projectUser == null) {
>                 throw IlleganArguementException("project user is null");
>         }

The DS project contains a large number of assert statements—I assumed they were recommended for use.
If we don't check in advance, a NullPointerException (NPE) will occur.

Reply:
Only a few historical codes contain assert statements. 

The proper fix is to set the property to empty, rather than ignore the alert.